### PR TITLE
Enable  clang-tidy on torch/csrc/distributed/autograd/*

### PR DESCRIPTION
**Pull request by cyyever**
Enable clang-tidy on `torch/csrc/distributed/autograd/*` directory.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: .lintrunner.toml [hunk 1]
```diff
@@ -216,6 +216,8 @@ include_patterns = [
     'torch/csrc/*.cpp',
     'torch/csrc/**/*.h',
     'torch/csrc/**/*.cpp',
```
Comment (Skylion007) [inline .lintrunner.toml:218]: Hmm, quick question why doesn't
'torch/csrc/**/*.cpp' also cover  `'torch/csrc/distributed/autograd/*.cpp'`, smells like a bug where `**` isn't being expanded properly by lintrunner maybe?

Comment (Skylion007) [reply .lintrunner.toml:218]: I think most of these issues are just because clang-tidy isn't new enough.

Comment (cyyever) [reply .lintrunner.toml:218]: > I think most of these issues are just because clang-tidy isn't new enough.

I had a PR to use clangtidy-18

**Pull request closed without merging**